x-pack/plugins/alerting/server/alerts_service/alerts_service.ts

```diff
@@ -146,70 +167,84 @@ export class AlertsService implements IAlertsService {
   }
 
   public register(opts: IRuleTypeAlerts, timeoutMs?: number) {
-    const { context, fieldMap } = opts;
+    const { context } = opts;
     // check whether this context has been registered before
     if (this.registeredContexts.has(context)) {
-      const registeredFieldMap = this.registeredContexts.get(context);
-      if (!isEqual(fieldMap, registeredFieldMap)) {
-        throw new Error(`${context} has already been registered with a different mapping`);
+      const registeredOptions = this.registeredContexts.get(context);
+      if (!isEqual(opts, registeredOptions)) {
```

Comment:
Testing equality of all options to catch the possibility of rule types registering with the same context but different `useEcs` or `useLegacyAlerts` flags.